### PR TITLE
Fix macOS CI

### PR DESCRIPTION
**Pull request by dave-tucker**
This PR contains 3 commits to fix the macOS CI job which has been failing since ~12th February.

The causes was that Rust nightly updated to LLVM v18.1-RC2 but bpf-linker was linked against LLVM 17. Even with the bpf-linker v0.9.11 release (which supports the LLVM RC) macOS CI still fails due to LLVM 18.1-RC2 not being installable via `brew` to link against. Solving this with homebrew tap/bottles and/or binstalls of bpf-linker is the best long term solution. In the short term we revert to the last working rust nightly toolchain + bpf-linker v0.9.10

This was pretty hard to debug due to missing output from `cargo xtask integration-test`.
A commit in this PR provides a fix for that too which should make this easier to debug in future.

Note: At some point in CI we were also seeing errors about missing libelf. This seems to be a red herring 🐟 
Commits in this PR that added libelf to the brew dependencies have been reverted

### CODE REVIEW

**Comment by tamird** [comment]
Just opened #884 and then saw this. The true failure is coming from bpf-linker but we're not getting its logs (again).

**Review by vadorovsky — APPROVED** [review]

**Comment by vadorovsky** [comment]
@dave-tucker @tamird bpf-linker is fixed, but it uses LLVM 18 now. To fix the latest failure, we need to find some way to install LLVM 18 on macOS.

**Comment by tamird** [comment]
We shouldn't merge this - it doesn't fix anything. We should figure out why we aren't getting errors from bpf-linker.

**Review by vadorovsky — CHANGES_REQUESTED** [review]
on hold until we sort out the LLVM 18 on macOS thing

**Comment by tamird** [comment]
The first failing job was https://github.com/aya-rs/aya/actions/runs/7896427886/job/21550389236.

What I'm saying is that I don't see the bitcode error there, it's not clear why the job failed from the logs.

**Comment by tamird** [comment]
Yes, this is exactly what I'm saying: we can't see the actual error.

**Comment by tamird** [comment]
Nasty! We have the same code in `xtask/src/run.rs` that we should also fix.

File: .github/workflows/ci.yml
Comment (tamird) [inline .github/workflows/ci.yml:237]: we should remove this commit?

**Review by tamird — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (dave-tucker) [reply .github/workflows/ci.yml:241]: afaict we still need it - it was an implicit dependency of libbpf before but we didn't install it.
once macOS CI is green again (which i'm hoping my last commit should do) i'll do an experiment to see whether it is still required.

**Comment by tamird** [comment]
(reminder to update the PR title plz)

**Comment by dave-tucker** [comment]
Note: Clippy will fail until #885 has merged. We can merge either first, but one will be red (unless I incorporate the clippy fix here too).

File: .github/workflows/ci.yml
Comment (dave-tucker) [reply .github/workflows/ci.yml]: done.

File: .github/workflows/ci.yml
Comment (tamird) [reply .github/workflows/ci.yml:250]: please drop a note here when you learn the answer

Comment (tamird) [reply .github/workflows/ci.yml]: It logs a warning :)

**Comment by dave-tucker** [comment]
Merging to get CI back to green!

**Pull request merged**
